### PR TITLE
Make logits be pulled from last non pad token in the input sequences.

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Removed a copy in SetLogits since Logits are generated in place now.
Fix some unit tests, as logits passed to SetLogits() are now written to.

### CODE REVIEW

**Review by kunal-vaishnavi — APPROVED** [review]

File: src/search_cuda.cpp [hunk 1]
```diff
@@ -62,27 +57,12 @@ BeamSearch_Cuda::BeamSearch_Cuda(const GeneratorParams& params)
...
-  const float* current_logits = logits.data() + (input_length - 1) * params_.vocab_size;
+  // Apply log_softmax to the input logits
   for (int i = 0; i < batch_beam_size; i++) {
-    std::span<const float> source(current_logits, params_.vocab_size);
     std::span<float> target = next_token_scores_.subspan(i * params_.vocab_size, params_.vocab_size);
-    CudaCheck() == cudaMemcpyAsync(target.data(), source.data(), source.size_bytes(), cudaMemcpyDeviceToDevice, params_.cuda_stream);
-    current_logits += input_length * params_.vocab_size;
-
     cuda::Launch_log_softmax(target.data(), static_cast<int>(target.size()), params_.cuda_stream);
   }
```
Comment (yufenglee) [inline src/search_cuda.cpp:67]: The implementation of Launch_log_softmax is quite naive. The performance will be bad.

**Review by yufenglee — APPROVED** [review]
:shipit:

File: src/search_cuda.cpp (hunk 1 above)
Comment (yufenglee) [reply src/search_cuda.cpp:67]: The Launch_log_softmax does nothing:
[https://github.com/microsoft/onnxruntime-genai/blob/dbd10985e109cbf76f9f67f4e939c26eacc88288/src/search_cuda.cu#L41-L54](https://github.com/microsoft/onnxruntime-genai/blob/dbd10985e109cbf76f9f67f4e939c26eacc88288/src/search_cuda.cu#L41-L54)

Comment (yufenglee) [reply src/search_cuda.cpp:67]: We don't need it on GPU as the Sampling handles it.

Comment (yufenglee) [reply src/search_cuda.cpp:67]: [https://github.com/microsoft/onnxruntime-genai/blob/2addfc6f4e6c307a9329d548078003c2dabf15ff/src/cuda_sampling.cu#L582](https://github.com/microsoft/onnxruntime-genai/blob/2addfc6f4e6c307a9329d548078003c2dabf15ff/src/cuda_sampling.cu#L582)

Comment (RyanUnderhill) [reply src/search_cuda.cpp]: That's right, I'll remove these parts as they're not doing any work.
